torch/utils/checkpoint.py

```diff
@@ -371,7 +381,12 @@ def inner_unpack(packed):
                     torch.set_rng_state(fwd_cpu_state)
                     if had_cuda_in_fwd:
                         set_device_states(fwd_gpu_devices, fwd_gpu_states)
-                with torch.enable_grad(), torch.cuda.amp.autocast(had_autocast_in_fwd):
+
+                with (
```

Comment:
nit: Stylistically, if we are doing `with ctx1, ctx2, ctx3:` already, maybe we can include the existing `saved_tensors_hooks()` context in the same `with` statement to avoid the nested `with`.

Reply:
Does

```
with (a, b)
```

guarantee that a is entered before b?

Reply:
Yes. I believe so.
<img width="818" alt="Screen Shot 2022-07-20 at 1 41 30 PM" src="https://user-images.githubusercontent.com/31054793/180047496-3a68397e-908e-4f0e-97c2-ab3d96bbe29c.png">
https://docs.python.org/3/reference/compound_stmts.html#with

Reply:
I see. Actually appears that this syntax is limited to 3.10?

Reply:
I think for 3.10, they let you use parentheses like:
```
with (
    A() as a,
    B() as b,
):
    SUITE
```
Already in 3.1, you can do like:
```
with A() as a, B() as b:
    SUITE
```
(For multi-line, you need to include the `\` yourself. See the second answer here: https://stackoverflow.com/questions/3024925/create-a-with-block-on-several-context-managers)